src/clf/compoundV3/compoundV3Computer.js

```diff
@@ -0,0 +1,305 @@
+        console.log(error);
+    }
+}
+
+function computeProtocolWeightedCLF(protocolData) {
+    console.log('protocolData', protocolData);
+    let protocolCollateral = 0;
+    const weightMap = {};
+    for (const [k, v] of Object.entries(protocolData)) {
+        console.log(k, v);
+        if (v) {
+            protocolCollateral += v['totalCollateral'];
+        }
+    }
+    // get each collateral weight
+    for (const [k, v] of Object.entries(protocolData)) {
+        console.log('second loop', k, v)
+        if (v) {
+            const weight = v['totalCollateral'] / protocolCollateral;
+            const clf = v['weightedCLF'];
+            weightMap[k] = weight * clf;
+        }
+    }
+    let weightedCLF = 0;
+    for (const [k, v] of Object.entries(weightMap)) {
+        weightedCLF += v;
+    }
+    weightedCLF = (weightedCLF).toFixed(2)
+    return weightedCLF;
+}
+
+async function computeCLFForPool(pool) {
+    const resultsData = {};
+    resultsData['data'] = {};
+
+    console.log(`Started work on Compound v3 --- ${pool.baseAsset} --- pool`);
+    const web3Provider = new ethers.providers.StaticJsonRpcProvider(process.env.RPC_URL);
+    const cometContract = new ethers.Contract(pool.cometAddress, cometABI, web3Provider);
+    /// for all collaterals in selected pool
+    for (const collateral of Object.values(pool.collateralTokens)) {
+        try {
+            console.log(`Computing CLFs for ${collateral.symbol}`);
+            resultsData['data'][collateral.symbol] = {};
+            resultsData['data'][collateral.symbol]['collateral'] = await getCollateralAmount(collateral, cometContract);
+            resultsData['data'][collateral.symbol]['clfs'] = await computeMarketCLF(web3Provider, cometContract, collateral, pool.baseAsset);
+            console.log('---------------------------');
+            console.log('---------------------------');
+            console.log('resultsData', resultsData);
+            console.log('---------------------------');
+            console.log('---------------------------');
+        }
+        catch (error) {
+            console.log('error', error);
+            resultsData['data'][collateral.symbol] = null;
+        }
+    }
+    return resultsData;
+}
+
+async function getCollateralAmount(collateral, cometContract) {
+    const [totalSupplyAsset] = await cometContract.callStatic.totalsCollateral(collateral.address);
+    const decimals = tokens[collateral.symbol.toUpperCase()].decimals;
+    const results = {};
+    let price = undefined;
+    try {
+        price = await axios.get(`https://api.coingecko.com/api/v3/simple/price?ids=${collateral.coinGeckoID}&vs_currencies=usd`);
+    }
+    catch (error) {
+        console.log('error fetching price', error);
+    }
+    price = price.data[collateral.coinGeckoID]['usd'];
+    results['inKindSupply'] = normalize(totalSupplyAsset, decimals);
+    results['usdSupply'] = results['inKindSupply'] * price;
+    return results;
```

Comment:
Here, if price is undefined (because of an exception), you still try to read price.data.